src/shipPossibilities.ts

```diff
@@ -0,0 +1,204 @@
+            console.log(currentRow);
+        }
+
+        console.log("vertical:");
+
+        for(var i=0;i <Globals.BOARD_ROWS; i++){
+            let currentRow:string = '';
+            for(var j = 0; j<Globals.BOARD_COLS; j++){
+                if(this.possibilities[i].row[j].vertical){
+                    currentRow += 'X';
+                }
+                else{
+                    currentRow += 'O';
+                }
+            }
+            console.log(currentRow);
+        }
+    }
+
+    private removeVerticalPossibilities(_startRow:number, endRow:number, startCol:number, endCol:number){
+        let startRow:number = Math.max(0,_startRow-(this.length-1));
+
+        for(var i=startRow;i <=endRow; i++){
+            for(var j = startCol; j<=endCol; j++){
+                if(this.possibilities[i].row[j].vertical)
+                {
+                    this.possibilities[i].row[j].vertical = false;
+                    this.possibilities[i].numInRow--;
+                    this.numberOfPossibilities--;
+                }
+            }
+        }
+    }
+
+    private removeHorizontalPossibilities(startRow:number, endRow:number, _startCol:number, endCol:number){
+        let startCol:number = Math.max(0,_startCol-(this.length-1));
+
+        for(var i=startRow;i <=endRow; i++){
+            for(var j = startCol; j<=endCol; j++){
+                if(this.possibilities[i].row[j].horizontal)
+                {
+                    this.possibilities[i].row[j].horizontal = false;
+                    this.possibilities[i].numInRow--;
+                    this.numberOfPossibilities--;
+                }
+            }
+        }
+    }
+
+
+    static generateAllPossibilities(length:number):{configs: ConfigurationSet, size:number}{
+        //initialize configSet
+        let currentPossibilities: ConfigurationSet = [];
+        for(var i=0; i<Globals.BOARD_ROWS; i++){
+            currentPossibilities.push({row: [],numInRow:0});
+            for(var j=0; j<Globals.BOARD_COLS; j++){
+                currentPossibilities[i].row.push({horizontal: false,vertical: false});
+            }
+        }
+
+        let count:number =0;
+
+        for(var i=0; i<Globals.BOARD_ROWS-length+1; i++){  
+            for(var j=0; j<Globals.BOARD_COLS; j++){
+                currentPossibilities[i].row[j].vertical = true;
+                currentPossibilities[i].numInRow++;
+                count++;
+            }
+        }
+
+        for(var i=0; i<Globals.BOARD_ROWS; i++){
+            for(var j=0; j<Globals.BOARD_COLS-length+1; j++){
+                currentPossibilities[i].row[j].horizontal = true;
+                currentPossibilities[i].numInRow++;
+                count++;
+            }
+        }
+        return {configs: currentPossibilities,size: count};
+    }
+
+    public pickRandomPlacement():ShipPlacement{
+        //get random number up to possibility count
+        //choose that one from the list
+        //generate the thing
```

Comment:
To be honest, I don't think you need these comments - this explains itself fine.